lib/Sema/ResilienceDiagnostics.cpp

```diff
@@ -127,9 +127,10 @@ bool TypeChecker::diagnoseInlinableDeclRefAccess(SourceLoc loc,
   if (D->getDeclContext()->isLocalContext())
     return false;
 
-  // Public declarations are OK.
+  // Public non-SPI declarations are OK.
```

Comment:
Do we diagnose @spi decl inside a non-resilient type marked of `@frozen`? it could be disallowed.

Reply:
This should be diagnosed. There's a test in `local_spi_decls.swift` but it could be extended to cover more edge cases.

Reply:
Sorry, I should be more specific here. Say when we have a frozen struct with three fields and the second field is marked as SPI. This means in the API world, the third field will be referenced as the second field in runtime even the user thinks they are referencing the third field by name. This could lead to unintended runtime behavior. I wonder whether this pattern is diagnosed in `local_spi_decls.swift`.